python/utils.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import os
+# Get the absolute path of the directory containing the .so file
+dir_path = os.path.dirname(os.path.abspath(__file__)) + "/../cpp"
+# Add the directory to the Python path
+sys.path.append(dir_path)
+
+import numpy as np
+from scipy.optimize import line_search
+from autograd import grad
+import autograd.numpy as au
+
+def is_pos_def(x):
+    return np.all(np.linalg.eigvals(x) > 0)
+
+def generate_pos_def(n):
+    A = np.random.rand(n, n)
+    return A.dot(A.T)
+
+def generate_symmetric(n):
+    A = np.random.rand(n, n)
+    return (A + A.T)/2
+
+def linear_CG(x, A, b, epsilon, epoch=10000000):
```

Comment:
You can make an inline comment in PR to make the discussions clearer.

Reply:
Thank you for notice, I will add the comment after Wednesday.